demo/default.nix

```diff
@@ -1,6 +1,23 @@
 # A shell setup providing build tools and utilities for the demo
 let
   project = import ../default.nix { };
+  nix-thunk = import ./dep/nix-thunk { inherit (project) pkgs; };
```

Comment:
What is nix-thunk?

Reply:
https://github.com/obsidiansystems/nix-thunk

Essentially a nice way of managing and fixing repos, while keeping them transparent to nix. 

I created the thunk to fix the issue, but started by trying newer versions, this version worked.